In [37]:
import pandas as pd

In [38]:
dados = pd.read_csv("State_of_data_2022.csv", sep=",")

In [39]:
dados.head()

('P0', 'id')  ('P1_a ', 'Idade')  \
0  zzqzz3l9ily8nuo2m7wyzzqzz3w48o96                39.0   
1  zzls2oftfn9law393oezzls2ofhvfpzd                32.0   
2  zzdwqzfqqp1ypc7ps6m0hzzdwqz292yi                53.0   
3  zzbqh3uy7yk7k9qmkzzbqtb4s9faqspl                27.0   
4  zzaf1m95yan929rb94wzzaf1mekhvhpg                46.0   

  ('P1_a_1 ', 'Faixa idade') ('P1_b ', 'Genero') ('P1_c ', 'Cor/raca/etnia')  \
0                      35-39           Masculino                       Parda   
1                      30-34           Masculino                       Parda   
2                      50-54           Masculino                      Branca   
3                      25-29           Masculino                      Branca   
4                      45-49            Feminino                      Branca   

  ('P1_d ', 'PCD')  ('P1_e ', 'experiencia_profissional_prejudicada')  \
0              Não  Não acredito que minha experiência profissiona...   
1              Não  Sim, acredito que a minha a experiência profis...   
2              Não                                                NaN   
3              Não                                                NaN   
4              Não  Não acredito que minha experiência profissiona...   

             ('P1_f ', 'aspectos_prejudicados')  ('P1_g ', 'vive_no_brasil')  \
0                                           NaN                         True   
1  Aprovação em processos seletivos/entrevistas                         True   
2                                           NaN                         True   
3                                           NaN                         True   
4                                           NaN                         True   

  ('P1_i ', 'Estado onde mora')  ...  \
0         Distrito Federal (DF)  ...   
1                     Pará (PA)  ...   
2         Distrito Federal (DF)  ...   
3             Minas Gerais (MG)  ...   
4                     Pará (PA)  ...   

  ('P8_d_2 ', 'Coletando e limpando os dados que uso para análise e modelagem.')  \
0                                                NaN                               
1                                                NaN                               
2                                                0.0                               
3                                                1.0                               
4                                                NaN                               

  ('P8_d_3 ', 'Entrando em contato com os times de negócio para definição do problema, identificar a solução e apresentação de resultados.')  \
0                                                NaN                                                                                           
1                                                NaN                                                                                           
2                                                1.0                                                                                           
3                                                0.0                                                                                           
4                                                NaN                                                                                           

  ('P8_d_4 ', 'Desenvolvendo modelos de Machine Learning com o objetivo de colocar em produção em sistemas (produtos de dados).')  \
0                                                NaN                                                                                
1                                                NaN                                                                                
2                                                0.0                                                                                
3                                                0.0                                                                        

In [40]:
lista = dados.columns.tolist()

divisao = {
    'P0': ('Parte0', 'Identificador'),
    'P1': ('Parte1', 'Dados demográficos'),
    'P2': ('Parte2', 'Dados sobre carreira'),
    'P3': ('Parte3', 'Desafios dos gestores de times de dados'),
    'P4': ('Parte4', 'Conhecimentos na área de dados'),
    'P5': ('Parte5', 'Objetivos na área de dados'),
    'P6': ('Parte6', 'Conhecimentos em Engenharia de Dados/DE'),
    'P7': ('Parte7', 'Conhecimentos em Análise de Dados/DA'),
    'P8': ('Parte8', 'Conhecimentos em Ciências de Dados/DS')
}

# Extrair os dados da lista
dados_extratos = [(item.split("'")[1], item.split("'")[3]) for item in lista]

# Criar DataFrame
descricao = pd.DataFrame(dados_extratos, columns=['IdentificadorQuestao', 'Questao'])

# Adicionar coluna de divisão do questionário
descricao['DivisaoQuestionario'] = descricao['IdentificadorQuestao'].apply(lambda x: divisao.get(x.split('_')[0], ('Outro', 'Outro'))[0])

# Adicionar coluna de identificador
descricao['NomeCategoria'] = descricao['IdentificadorQuestao'].apply(lambda x: divisao.get(x.split('_')[0], ('Outro', 'Outro'))[1])

# Reorganizar colunas
descricao = descricao[['DivisaoQuestionario', 'NomeCategoria', 'IdentificadorQuestao', 'Questao']]

# Exibir o DataFrame
descricao.head()


DivisaoQuestionario       NomeCategoria IdentificadorQuestao         Questao
0              Parte0       Identificador                   P0              id
1              Parte1  Dados demográficos                P1_a            Idade
2              Parte1  Dados demográficos              P1_a_1      Faixa idade
3              Parte1  Dados demográficos                P1_b           Genero
4              Parte1  Dados demográficos                P1_c   Cor/raca/etnia

In [41]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte1']
categoria

DivisaoQuestionario       NomeCategoria IdentificadorQuestao  \
1               Parte1  Dados demográficos                P1_a    
2               Parte1  Dados demográficos              P1_a_1    
3               Parte1  Dados demográficos                P1_b    
4               Parte1  Dados demográficos                P1_c    
5               Parte1  Dados demográficos                P1_d    
6               Parte1  Dados demográficos                P1_e    
7               Parte1  Dados demográficos                P1_f    
8               Parte1  Dados demográficos                P1_g    
9               Parte1  Dados demográficos                P1_i    
10              Parte1  Dados demográficos              P1_i_1    
11              Parte1  Dados demográficos              P1_i_2    
12              Parte1  Dados demográficos                P1_j    
13              Parte1  Dados demográficos                P1_k    
14              Parte1  Dados demográficos                P1_l    
15              Parte1  Dados demográficos                P1_m    

                                 Questao  
1                                  Idade  
2                            Faixa idade  
3                                 Genero  
4                         Cor/raca/etnia  
5                                    PCD  
6   experiencia_profissional_prejudicada  
7                  aspectos_prejudicados  
8                         vive_no_brasil  
9                       Estado onde mora  
10                          uf onde mora  
11                      Regiao onde mora  
12                      Mudou de Estado?  
13                      Regiao de origem  
14                       Nivel de Ensino  
15                      Área de Formação

In [42]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte2']
categoria

DivisaoQuestionario         NomeCategoria IdentificadorQuestao  \
16              Parte2  Dados sobre carreira                P2_a    
17              Parte2  Dados sobre carreira                P2_b    
18              Parte2  Dados sobre carreira                P2_c    
19              Parte2  Dados sobre carreira                P2_d    
20              Parte2  Dados sobre carreira                P2_e    
21              Parte2  Dados sobre carreira                P2_f    
22              Parte2  Dados sobre carreira                P2_g    
23              Parte2  Dados sobre carreira                P2_h    
24              Parte2  Dados sobre carreira                P2_i    
25              Parte2  Dados sobre carreira                P2_j    
26              Parte2  Dados sobre carreira                P2_k    
27              Parte2  Dados sobre carreira                P2_l    
28              Parte2  Dados sobre carreira              P2_l_1    
29              Parte2  Dados sobre carreira              P2_l_2    
30              Parte2  Dados sobre carreira              P2_l_3    
31              Parte2  Dados sobre carreira              P2_l_4    
32              Parte2  Dados sobre carreira              P2_l_5    
33              Parte2  Dados sobre carreira              P2_l_6    
34              Parte2  Dados sobre carreira              P2_l_7    
35              Parte2  Dados sobre carreira                P2_m    
36              Parte2  Dados sobre carreira                P2_n    
37              Parte2  Dados sobre carreira                P2_o    
38              Parte2  Dados sobre carreira              P2_o_1    
39              Parte2  Dados sobre carreira              P2_o_2    
40              Parte2  Dados sobre carreira              P2_o_3    
41              Parte2  Dados sobre carreira              P2_o_4    
42              Parte2  Dados sobre carreira              P2_o_5    
43              Parte2  Dados sobre carreira              P2_o_6    
44              Parte2  Dados sobre carreira              P2_o_7    
45              Parte2  Dados sobre carreira              P2_o_8    
46              Parte2  Dados sobre carreira              P2_o_9    
47              Parte2  Dados sobre carreira             P2_o_10    
48              Parte2  Dados sobre carreira                P2_p    
49              Parte2  Dados sobre carreira                P2_q    
50              Parte2  Dados sobre carreira                P2_r    
51              Parte2  Dados sobre carreira                P2_s    

                                              Questao  
16               Qual sua situação atual de trabalho?  
17                                              Setor  
18                             Numero de Funcionarios  
19                                            Gestor?  
20                                  Cargo como Gestor  
21                                        Cargo Atual  
22                                              Nivel  
23                                     Faixa salarial  
24  Quanto tempo de experiência na área de dados v...  
25  Quanto tempo de experiência na área de TI/Enge...  
26         Você está satisfeito na sua empresa atual?  
27  Qual o principal motivo da sua insatisfação co...  
28  Falta de oportunidade de crescimento no empreg...  
29           Salário atual não corresponde ao mercado  
30     Não tenho uma boa relação com meu líder/gestor  
31  Gostaria de trabalhar em em outra área de atuação  
32                Gostaria de receber mais benefícios  
33             O clima de trabalho/ambiente não é bom  
34           Falta de maturidade analítica na empresa  
35  Você participou de entrevistas de emprego nos ...  
36  Você pretende mudar de emprego nos próximos 6 ...  
37  Quais os principais critérios que você leva em...  
38                                Remuneração/Salário  
39                                         Benefícios  
40                 Propósito do trabalho e da emp

In [43]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte3']
categoria

DivisaoQuestionario                            NomeCategoria  \
52              Parte3  Desafios dos gestores de times de dados   
53              Parte3  Desafios dos gestores de times de dados   
54              Parte3  Desafios dos gestores de times de dados   
55              Parte3  Desafios dos gestores de times de dados   
56              Parte3  Desafios dos gestores de times de dados   
57              Parte3  Desafios dos gestores de times de dados   
58              Parte3  Desafios dos gestores de times de dados   
59              Parte3  Desafios dos gestores de times de dados   
60              Parte3  Desafios dos gestores de times de dados   
61              Parte3  Desafios dos gestores de times de dados   
62              Parte3  Desafios dos gestores de times de dados   
63              Parte3  Desafios dos gestores de times de dados   
64              Parte3  Desafios dos gestores de times de dados   
65              Parte3  Desafios dos gestores de times de dados   
66              Parte3  Desafios dos gestores de times de dados   
67              Parte3  Desafios dos gestores de times de dados   
68              Parte3  Desafios dos gestores de times de dados   
69              Parte3  Desafios dos gestores de times de dados   
70              Parte3  Desafios dos gestores de times de dados   
71              Parte3  Desafios dos gestores de times de dados   
72              Parte3  Desafios dos gestores de times de dados   
73              Parte3  Desafios dos gestores de times de dados   
74              Parte3  Desafios dos gestores de times de dados   
75              Parte3  Desafios dos gestores de times de dados   
76              Parte3  Desafios dos gestores de times de dados   
77              Parte3  Desafios dos gestores de times de dados   
78              Parte3  Desafios dos gestores de times de dados   
79              Parte3  Desafios dos gestores de times de dados   
80              Parte3  Desafios dos gestores de times de dados   
81              Parte3  Desafios dos gestores de times de dados   
82              Parte3  Desafios dos gestores de times de dados   
83              Parte3  Desafios dos gestores de times de dados   
84              Parte3  Desafios dos gestores de times de dados   
85              Parte3  Desafios dos gestores de times de dados   
86              Parte3  Desafios dos gestores de times de dados   
87              Parte3  Desafios dos gestores de times de dados   
88              Parte3  Desafios dos gestores de times de dados   
89              Parte3  Desafios dos gestores de times de dados   

   IdentificadorQuestao                                            Questao  
52                P3_a   Qual o número aproximado de pessoas que atuam ...  
53                P3_b   Quais desses papéis/cargos fazem parte do time...  
54              P3_b_1                                  Analytics Engineer  
55              P3_b_2                   Engenharia de Dados/Data Engineer  
56              P3_b_3                      Analista de Dados/Data Analyst  
57              P3_b_4                   Cientista de Dados/Data Scientist  
58              P3_b_5                          Database Administrator/DBA  
59              P3_b_6                Analista de Business Intelligence/BI  
60              P3_b_7                   Arquiteto de Dados/Data Architect  
61              P3_b_8                            Data Product Manager/DPM  
62              P3_b_9                                    Business Analyst  
63                P3_c   Quais dessas responsabilidades fazem parte da ...  
64              P3_c_1   Pensar na visão de longo prazo de dados da emp...  
65              P3_c_2   Organização de treinamentos e iniciativas com ...  
66              P3_c_3   Atração, seleção e contratação de talentos par...  
67              P3_c_4   Decisão sobre contratação de ferramentas e tec...  
68              P3_c_5   Sou gestor da equipe responsável pela engenhar...  
69 

In [44]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte4']
categoria

DivisaoQuestionario                   NomeCategoria IdentificadorQuestao  \
90               Parte4  Conhecimentos na área de dados                P4_a    
91               Parte4  Conhecimentos na área de dados              P4_a_1    
92               Parte4  Conhecimentos na área de dados                P4_b    
93               Parte4  Conhecimentos na área de dados              P4_b_1    
94               Parte4  Conhecimentos na área de dados              P4_b_2    
..                  ...                             ...                  ...   
185              Parte4  Conhecimentos na área de dados             P4_i_19    
186              Parte4  Conhecimentos na área de dados             P4_i_20    
187              Parte4  Conhecimentos na área de dados             P4_i_21    
188              Parte4  Conhecimentos na área de dados             P4_i_22    
189              Parte4  Conhecimentos na área de dados             P4_i_23    

                                               Questao  
90   Mesmo que esse não seja seu cargo formal, você...  
91                                             Atuacao  
92   Quais das fontes de dados listadas você já ana...  
93      Dados relacionais (estruturados em bancos SQL)  
94                   Dados armazenados em bancos NoSQL  
..                                                 ...  
185                                            Grafana  
186                                     TIBCO Spotfire  
187                                            Pentaho  
188  Fazemos todas as análises utilizando apenas Ex...  
189   Não utilizo nenhuma ferramenta de BI no trabalho  

[100 rows x 4 columns]

In [45]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte5']
categoria

DivisaoQuestionario               NomeCategoria IdentificadorQuestao  \
190              Parte5  Objetivos na área de dados                P5_a    
191              Parte5  Objetivos na área de dados                P5_b    
192              Parte5  Objetivos na área de dados                P5_c    
193              Parte5  Objetivos na área de dados                P5_d    

                                               Questao  
190                Qual seu objetivo na área de dados?  
191              Qual oportunidade você está buscando?  
192  Há quanto tempo você busca uma oportunidade na...  
193  Como tem sido a busca por um emprego na área d...

In [46]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte6']
categoria

DivisaoQuestionario                            NomeCategoria  \
194              Parte6  Conhecimentos em Engenharia de Dados/DE   
195              Parte6  Conhecimentos em Engenharia de Dados/DE   
196              Parte6  Conhecimentos em Engenharia de Dados/DE   
197              Parte6  Conhecimentos em Engenharia de Dados/DE   
198              Parte6  Conhecimentos em Engenharia de Dados/DE   
199              Parte6  Conhecimentos em Engenharia de Dados/DE   
200              Parte6  Conhecimentos em Engenharia de Dados/DE   
201              Parte6  Conhecimentos em Engenharia de Dados/DE   
202              Parte6  Conhecimentos em Engenharia de Dados/DE   
203              Parte6  Conhecimentos em Engenharia de Dados/DE   
204              Parte6  Conhecimentos em Engenharia de Dados/DE   
205              Parte6  Conhecimentos em Engenharia de Dados/DE   
206              Parte6  Conhecimentos em Engenharia de Dados/DE   
207              Parte6  Conhecimentos em Engenharia de Dados/DE   
208              Parte6  Conhecimentos em Engenharia de Dados/DE   
209              Parte6  Conhecimentos em Engenharia de Dados/DE   
210              Parte6  Conhecimentos em Engenharia de Dados/DE   
211              Parte6  Conhecimentos em Engenharia de Dados/DE   
212              Parte6  Conhecimentos em Engenharia de Dados/DE   
213              Parte6  Conhecimentos em Engenharia de Dados/DE   
214              Parte6  Conhecimentos em Engenharia de Dados/DE   
215              Parte6  Conhecimentos em Engenharia de Dados/DE   
216              Parte6  Conhecimentos em Engenharia de Dados/DE   
217              Parte6  Conhecimentos em Engenharia de Dados/DE   
218              Parte6  Conhecimentos em Engenharia de Dados/DE   
219              Parte6  Conhecimentos em Engenharia de Dados/DE   
220              Parte6  Conhecimentos em Engenharia de Dados/DE   
221              Parte6  Conhecimentos em Engenharia de Dados/DE   
222              Parte6  Conhecimentos em Engenharia de Dados/DE   
223              Parte6  Conhecimentos em Engenharia de Dados/DE   
224              Parte6  Conhecimentos em Engenharia de Dados/DE   
225              Parte6  Conhecimentos em Engenharia de Dados/DE   
226              Parte6  Conhecimentos em Engenharia de Dados/DE   
227              Parte6  Conhecimentos em Engenharia de Dados/DE   
228              Parte6  Conhecimentos em Engenharia de Dados/DE   
229              Parte6  Conhecimentos em Engenharia de Dados/DE   
230              Parte6  Conhecimentos em Engenharia de Dados/DE   
231              Parte6  Conhecimentos em Engenharia de Dados/DE   
232              Parte6  Conhecimentos em Engenharia de Dados/DE   
233              Parte6  Conhecimentos em Engenharia de Dados/DE   
234              Parte6  Conhecimentos em Engenharia de Dados/DE   
235              Parte6  Conhecimentos em Engenharia de Dados/DE   
236              Parte6  Conhecimentos em Engenharia de Dados/DE   
237              Parte6  Conhecimentos em Engenharia de Dados/DE   
238              Parte6  Conhecimentos em Engenharia de Dados/DE   
239              Parte6  Conhecimentos em Engenharia de Dados/DE   
240              Parte6  Conhecimentos em Engenharia de Dados/DE   
241              Parte6  Conhecimentos em Engenharia de Dados/DE   
242              Parte6  Conhecimentos em Engenharia de Dados/DE   
243              Parte6  Conhecimentos em Engenharia de Dados/DE   
244              Parte6  Conhecimentos em Engenharia de Dados/DE   
245              Parte6  Conhecimentos em Engenharia de Dados/DE   
246              Parte6  Conhecimentos em Engenharia de Dados/DE   
247              Parte6  Conhecimentos em Engenharia de Dados/DE   
248              Parte6  Conhecimentos em Engenharia de Dados/DE   
249              Parte6  Conhecimentos em Engenharia de Dados/DE   
250              Parte6  Conhecimentos em Engenharia de Dados/DE   
251              Parte6  Conhecimentos em Engenharia de Dado

In [47]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte7']
categoria

DivisaoQuestionario                         NomeCategoria  \
253              Parte7  Conhecimentos em Análise de Dados/DA   
254              Parte7  Conhecimentos em Análise de Dados/DA   
255              Parte7  Conhecimentos em Análise de Dados/DA   
256              Parte7  Conhecimentos em Análise de Dados/DA   
257              Parte7  Conhecimentos em Análise de Dados/DA   
258              Parte7  Conhecimentos em Análise de Dados/DA   
259              Parte7  Conhecimentos em Análise de Dados/DA   
260              Parte7  Conhecimentos em Análise de Dados/DA   
261              Parte7  Conhecimentos em Análise de Dados/DA   
262              Parte7  Conhecimentos em Análise de Dados/DA   
263              Parte7  Conhecimentos em Análise de Dados/DA   
264              Parte7  Conhecimentos em Análise de Dados/DA   
265              Parte7  Conhecimentos em Análise de Dados/DA   
266              Parte7  Conhecimentos em Análise de Dados/DA   
267              Parte7  Conhecimentos em Análise de Dados/DA   
268              Parte7  Conhecimentos em Análise de Dados/DA   
269              Parte7  Conhecimentos em Análise de Dados/DA   
270              Parte7  Conhecimentos em Análise de Dados/DA   
271              Parte7  Conhecimentos em Análise de Dados/DA   
272              Parte7  Conhecimentos em Análise de Dados/DA   
273              Parte7  Conhecimentos em Análise de Dados/DA   
274              Parte7  Conhecimentos em Análise de Dados/DA   
275              Parte7  Conhecimentos em Análise de Dados/DA   
276              Parte7  Conhecimentos em Análise de Dados/DA   
277              Parte7  Conhecimentos em Análise de Dados/DA   
278              Parte7  Conhecimentos em Análise de Dados/DA   
279              Parte7  Conhecimentos em Análise de Dados/DA   
280              Parte7  Conhecimentos em Análise de Dados/DA   
281              Parte7  Conhecimentos em Análise de Dados/DA   
282              Parte7  Conhecimentos em Análise de Dados/DA   
283              Parte7  Conhecimentos em Análise de Dados/DA   
284              Parte7  Conhecimentos em Análise de Dados/DA   
285              Parte7  Conhecimentos em Análise de Dados/DA   
286              Parte7  Conhecimentos em Análise de Dados/DA   
287              Parte7  Conhecimentos em Análise de Dados/DA   
288              Parte7  Conhecimentos em Análise de Dados/DA   
289              Parte7  Conhecimentos em Análise de Dados/DA   
290              Parte7  Conhecimentos em Análise de Dados/DA   
291              Parte7  Conhecimentos em Análise de Dados/DA   
292              Parte7  Conhecimentos em Análise de Dados/DA   
293              Parte7  Conhecimentos em Análise de Dados/DA   
294              Parte7  Conhecimentos em Análise de Dados/DA   
295              Parte7  Conhecimentos em Análise de Dados/DA   
296              Parte7  Conhecimentos em Análise de Dados/DA   
297              Parte7  Conhecimentos em Análise de Dados/DA   
298              Parte7  Conhecimentos em Análise de Dados/DA   
299              Parte7  Conhecimentos em Análise de Dados/DA   
300              Parte7  Conhecimentos em Análise de Dados/DA   
301              Parte7  Conhecimentos em Análise de Dados/DA   
302              Parte7  Conhecimentos em Análise de Dados/DA   

    IdentificadorQuestao                                            Questao  
253                P7_1   Quais das opções abaixo fazem parte da sua rot...  
254              P7_a_1   Processo e analiso dados utilizando linguagens...  
255              P7_a_2   Realizo construções de dashboards em ferrament...  
256              P7_a_3   Crio consultas através da linguagem SQL para e...  
257              P7_a_4                                         Utilizo API  
258              P7_a_5   Realizo experimentos e estudos utilizando meto...  
259              P7_a_6               Desenvolvo/cuido da manutenção de ETL  
260              P7_a_7   Atuo na modelagem dos dados, com o obj

In [48]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte8']
categoria

DivisaoQuestionario                          NomeCategoria  \
303              Parte8  Conhecimentos em Ciências de Dados/DS   
304              Parte8  Conhecimentos em Ciências de Dados/DS   
305              Parte8  Conhecimentos em Ciências de Dados/DS   
306              Parte8  Conhecimentos em Ciências de Dados/DS   
307              Parte8  Conhecimentos em Ciências de Dados/DS   
308              Parte8  Conhecimentos em Ciências de Dados/DS   
309              Parte8  Conhecimentos em Ciências de Dados/DS   
310              Parte8  Conhecimentos em Ciências de Dados/DS   
311              Parte8  Conhecimentos em Ciências de Dados/DS   
312              Parte8  Conhecimentos em Ciências de Dados/DS   
313              Parte8  Conhecimentos em Ciências de Dados/DS   
314              Parte8  Conhecimentos em Ciências de Dados/DS   
315              Parte8  Conhecimentos em Ciências de Dados/DS   
316              Parte8  Conhecimentos em Ciências de Dados/DS   
317              Parte8  Conhecimentos em Ciências de Dados/DS   
318              Parte8  Conhecimentos em Ciências de Dados/DS   
319              Parte8  Conhecimentos em Ciências de Dados/DS   
320              Parte8  Conhecimentos em Ciências de Dados/DS   
321              Parte8  Conhecimentos em Ciências de Dados/DS   
322              Parte8  Conhecimentos em Ciências de Dados/DS   
323              Parte8  Conhecimentos em Ciências de Dados/DS   
324              Parte8  Conhecimentos em Ciências de Dados/DS   
325              Parte8  Conhecimentos em Ciências de Dados/DS   
326              Parte8  Conhecimentos em Ciências de Dados/DS   
327              Parte8  Conhecimentos em Ciências de Dados/DS   
328              Parte8  Conhecimentos em Ciências de Dados/DS   
329              Parte8  Conhecimentos em Ciências de Dados/DS   
330              Parte8  Conhecimentos em Ciências de Dados/DS   
331              Parte8  Conhecimentos em Ciências de Dados/DS   
332              Parte8  Conhecimentos em Ciências de Dados/DS   
333              Parte8  Conhecimentos em Ciências de Dados/DS   
334              Parte8  Conhecimentos em Ciências de Dados/DS   
335              Parte8  Conhecimentos em Ciências de Dados/DS   
336              Parte8  Conhecimentos em Ciências de Dados/DS   
337              Parte8  Conhecimentos em Ciências de Dados/DS   
338              Parte8  Conhecimentos em Ciências de Dados/DS   
339              Parte8  Conhecimentos em Ciências de Dados/DS   
340              Parte8  Conhecimentos em Ciências de Dados/DS   
341              Parte8  Conhecimentos em Ciências de Dados/DS   
342              Parte8  Conhecimentos em Ciências de Dados/DS   
343              Parte8  Conhecimentos em Ciências de Dados/DS   
344              Parte8  Conhecimentos em Ciências de Dados/DS   
345              Parte8  Conhecimentos em Ciências de Dados/DS   
346              Parte8  Conhecimentos em Ciências de Dados/DS   
347              Parte8  Conhecimentos em Ciências de Dados/DS   
348              Parte8  Conhecimentos em Ciências de Dados/DS   
349              Parte8  Conhecimentos em Ciências de Dados/DS   
350              Parte8  Conhecimentos em Ciências de Dados/DS   
351              Parte8  Conhecimentos em Ciências de Dados/DS   
352              Parte8  Conhecimentos em Ciências de Dados/DS   

    IdentificadorQuestao                                            Questao  
303                P8_a   Quais das opções abaixo fazem parte da sua rot...  
304              P8_a_1   Estudos Ad-hoc com o objetivo de confirmar hip...  
305              P8_a_2   Sou responsável pela coleta e limpeza dos dado...  
306              P8_a_3   Sou responsável por entrar em contato com os t...  
307              P8_a_4   Desenvolvo modelos de Machine Learning com o o...  
308              P8_a_5   Sou responsável por colocar modelos em produçã...  
309              P8_a_6   Cuido da manutenção de modelos de Machine Lear...  
310          

# Fazendo coluna pergunta

In [49]:
def extrair_pergunta(identificador):
    partes = identificador.split('_')
    if len(partes) > 2:
        return '_'.join(partes[:2])
    else:
        return identificador

# Aplicando a função à coluna "IdentificadorQuestao" para criar a nova coluna "pergunta"
descricao.loc[:, 'pergunta'] = descricao['IdentificadorQuestao'].apply(extrair_pergunta)

In [50]:
descricao.head()

DivisaoQuestionario       NomeCategoria IdentificadorQuestao  \
0              Parte0       Identificador                   P0   
1              Parte1  Dados demográficos                P1_a    
2              Parte1  Dados demográficos              P1_a_1    
3              Parte1  Dados demográficos                P1_b    
4              Parte1  Dados demográficos                P1_c    

          Questao pergunta  
0              id       P0  
1           Idade    P1_a   
2     Faixa idade     P1_a  
3          Genero    P1_b   
4  Cor/raca/etnia    P1_c

In [51]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte8']
categoria

DivisaoQuestionario                          NomeCategoria  \
303              Parte8  Conhecimentos em Ciências de Dados/DS   
304              Parte8  Conhecimentos em Ciências de Dados/DS   
305              Parte8  Conhecimentos em Ciências de Dados/DS   
306              Parte8  Conhecimentos em Ciências de Dados/DS   
307              Parte8  Conhecimentos em Ciências de Dados/DS   
308              Parte8  Conhecimentos em Ciências de Dados/DS   
309              Parte8  Conhecimentos em Ciências de Dados/DS   
310              Parte8  Conhecimentos em Ciências de Dados/DS   
311              Parte8  Conhecimentos em Ciências de Dados/DS   
312              Parte8  Conhecimentos em Ciências de Dados/DS   
313              Parte8  Conhecimentos em Ciências de Dados/DS   
314              Parte8  Conhecimentos em Ciências de Dados/DS   
315              Parte8  Conhecimentos em Ciências de Dados/DS   
316              Parte8  Conhecimentos em Ciências de Dados/DS   
317              Parte8  Conhecimentos em Ciências de Dados/DS   
318              Parte8  Conhecimentos em Ciências de Dados/DS   
319              Parte8  Conhecimentos em Ciências de Dados/DS   
320              Parte8  Conhecimentos em Ciências de Dados/DS   
321              Parte8  Conhecimentos em Ciências de Dados/DS   
322              Parte8  Conhecimentos em Ciências de Dados/DS   
323              Parte8  Conhecimentos em Ciências de Dados/DS   
324              Parte8  Conhecimentos em Ciências de Dados/DS   
325              Parte8  Conhecimentos em Ciências de Dados/DS   
326              Parte8  Conhecimentos em Ciências de Dados/DS   
327              Parte8  Conhecimentos em Ciências de Dados/DS   
328              Parte8  Conhecimentos em Ciências de Dados/DS   
329              Parte8  Conhecimentos em Ciências de Dados/DS   
330              Parte8  Conhecimentos em Ciências de Dados/DS   
331              Parte8  Conhecimentos em Ciências de Dados/DS   
332              Parte8  Conhecimentos em Ciências de Dados/DS   
333              Parte8  Conhecimentos em Ciências de Dados/DS   
334              Parte8  Conhecimentos em Ciências de Dados/DS   
335              Parte8  Conhecimentos em Ciências de Dados/DS   
336              Parte8  Conhecimentos em Ciências de Dados/DS   
337              Parte8  Conhecimentos em Ciências de Dados/DS   
338              Parte8  Conhecimentos em Ciências de Dados/DS   
339              Parte8  Conhecimentos em Ciências de Dados/DS   
340              Parte8  Conhecimentos em Ciências de Dados/DS   
341              Parte8  Conhecimentos em Ciências de Dados/DS   
342              Parte8  Conhecimentos em Ciências de Dados/DS   
343              Parte8  Conhecimentos em Ciências de Dados/DS   
344              Parte8  Conhecimentos em Ciências de Dados/DS   
345              Parte8  Conhecimentos em Ciências de Dados/DS   
346              Parte8  Conhecimentos em Ciências de Dados/DS   
347              Parte8  Conhecimentos em Ciências de Dados/DS   
348              Parte8  Conhecimentos em Ciências de Dados/DS   
349              Parte8  Conhecimentos em Ciências de Dados/DS   
350              Parte8  Conhecimentos em Ciências de Dados/DS   
351              Parte8  Conhecimentos em Ciências de Dados/DS   
352              Parte8  Conhecimentos em Ciências de Dados/DS   

    IdentificadorQuestao                                            Questao  \
303                P8_a   Quais das opções abaixo fazem parte da sua rot...   
304              P8_a_1   Estudos Ad-hoc com o objetivo de confirmar hip...   
305              P8_a_2   Sou responsável pela coleta e limpeza dos dado...   
306              P8_a_3   Sou responsável por entrar em contato com os t...   
307              P8_a_4   Desenvolvo modelos de Machine Learning com o o...   
308              P8_a_5   Sou responsável por colocar modelos em produçã...   
309              P8_a_6   Cuido da manutenção de modelos de Machine Lear...   
310  

# Notei um problema aqui

In [52]:
descricao_filtrada = descricao[descricao['IdentificadorQuestao'].str.len() <= 5]
descricao_filtrada

DivisaoQuestionario                          NomeCategoria  \
0                Parte0                          Identificador   
1                Parte1                     Dados demográficos   
3                Parte1                     Dados demográficos   
4                Parte1                     Dados demográficos   
5                Parte1                     Dados demográficos   
..                  ...                                    ...   
292              Parte7   Conhecimentos em Análise de Dados/DA   
303              Parte8  Conhecimentos em Ciências de Dados/DS   
315              Parte8  Conhecimentos em Ciências de Dados/DS   
329              Parte8  Conhecimentos em Ciências de Dados/DS   
341              Parte8  Conhecimentos em Ciências de Dados/DS   

    IdentificadorQuestao                                            Questao  \
0                     P0                                                 id   
1                  P1_a                                               Idade   
3                  P1_b                                              Genero   
4                  P1_c                                      Cor/raca/etnia   
5                  P1_d                                                 PCD   
..                   ...                                                ...   
292                P7_d   Em qual das opções abaixo você gasta a maior p...   
303                P8_a   Quais das opções abaixo fazem parte da sua rot...   
315                P8_b   Quais as técnicas e métodos listados abaixo vo...   
329                P8_3   Quais dessas tecnologias fazem parte do seu di...   
341                P8_d   Em qual das opções abaixo você gasta a maior p...   

    pergunta  
0         P0  
1      P1_a   
3      P1_b   
4      P1_c   
5      P1_d   
..       ...  
292    P7_d   
303    P8_a   
315    P8_b   
329    P8_3   
341    P8_d   

[66 rows x 5 columns]

In [53]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte1']
categoria

DivisaoQuestionario       NomeCategoria IdentificadorQuestao  \
1               Parte1  Dados demográficos                P1_a    
2               Parte1  Dados demográficos              P1_a_1    
3               Parte1  Dados demográficos                P1_b    
4               Parte1  Dados demográficos                P1_c    
5               Parte1  Dados demográficos                P1_d    
6               Parte1  Dados demográficos                P1_e    
7               Parte1  Dados demográficos                P1_f    
8               Parte1  Dados demográficos                P1_g    
9               Parte1  Dados demográficos                P1_i    
10              Parte1  Dados demográficos              P1_i_1    
11              Parte1  Dados demográficos              P1_i_2    
12              Parte1  Dados demográficos                P1_j    
13              Parte1  Dados demográficos                P1_k    
14              Parte1  Dados demográficos                P1_l    
15              Parte1  Dados demográficos                P1_m    

                                 Questao pergunta  
1                                  Idade    P1_a   
2                            Faixa idade     P1_a  
3                                 Genero    P1_b   
4                         Cor/raca/etnia    P1_c   
5                                    PCD    P1_d   
6   experiencia_profissional_prejudicada    P1_e   
7                  aspectos_prejudicados    P1_f   
8                         vive_no_brasil    P1_g   
9                       Estado onde mora    P1_i   
10                          uf onde mora     P1_i  
11                      Regiao onde mora     P1_i  
12                      Mudou de Estado?    P1_j   
13                      Regiao de origem    P1_k   
14                       Nivel de Ensino    P1_l   
15                      Área de Formação    P1_m

In [54]:
descricao_filtrada = categoria[categoria['IdentificadorQuestao'].str.len() <= 5]
descricao_filtrada

DivisaoQuestionario       NomeCategoria IdentificadorQuestao  \
1               Parte1  Dados demográficos                P1_a    
3               Parte1  Dados demográficos                P1_b    
4               Parte1  Dados demográficos                P1_c    
5               Parte1  Dados demográficos                P1_d    
6               Parte1  Dados demográficos                P1_e    
7               Parte1  Dados demográficos                P1_f    
8               Parte1  Dados demográficos                P1_g    
9               Parte1  Dados demográficos                P1_i    
12              Parte1  Dados demográficos                P1_j    
13              Parte1  Dados demográficos                P1_k    
14              Parte1  Dados demográficos                P1_l    
15              Parte1  Dados demográficos                P1_m    

                                 Questao pergunta  
1                                  Idade    P1_a   
3                                 Genero    P1_b   
4                         Cor/raca/etnia    P1_c   
5                                    PCD    P1_d   
6   experiencia_profissional_prejudicada    P1_e   
7                  aspectos_prejudicados    P1_f   
8                         vive_no_brasil    P1_g   
9                       Estado onde mora    P1_i   
12                      Mudou de Estado?    P1_j   
13                      Regiao de origem    P1_k   
14                       Nivel de Ensino    P1_l   
15                      Área de Formação    P1_m

In [55]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte2']
categoria

DivisaoQuestionario         NomeCategoria IdentificadorQuestao  \
16              Parte2  Dados sobre carreira                P2_a    
17              Parte2  Dados sobre carreira                P2_b    
18              Parte2  Dados sobre carreira                P2_c    
19              Parte2  Dados sobre carreira                P2_d    
20              Parte2  Dados sobre carreira                P2_e    
21              Parte2  Dados sobre carreira                P2_f    
22              Parte2  Dados sobre carreira                P2_g    
23              Parte2  Dados sobre carreira                P2_h    
24              Parte2  Dados sobre carreira                P2_i    
25              Parte2  Dados sobre carreira                P2_j    
26              Parte2  Dados sobre carreira                P2_k    
27              Parte2  Dados sobre carreira                P2_l    
28              Parte2  Dados sobre carreira              P2_l_1    
29              Parte2  Dados sobre carreira              P2_l_2    
30              Parte2  Dados sobre carreira              P2_l_3    
31              Parte2  Dados sobre carreira              P2_l_4    
32              Parte2  Dados sobre carreira              P2_l_5    
33              Parte2  Dados sobre carreira              P2_l_6    
34              Parte2  Dados sobre carreira              P2_l_7    
35              Parte2  Dados sobre carreira                P2_m    
36              Parte2  Dados sobre carreira                P2_n    
37              Parte2  Dados sobre carreira                P2_o    
38              Parte2  Dados sobre carreira              P2_o_1    
39              Parte2  Dados sobre carreira              P2_o_2    
40              Parte2  Dados sobre carreira              P2_o_3    
41              Parte2  Dados sobre carreira              P2_o_4    
42              Parte2  Dados sobre carreira              P2_o_5    
43              Parte2  Dados sobre carreira              P2_o_6    
44              Parte2  Dados sobre carreira              P2_o_7    
45              Parte2  Dados sobre carreira              P2_o_8    
46              Parte2  Dados sobre carreira              P2_o_9    
47              Parte2  Dados sobre carreira             P2_o_10    
48              Parte2  Dados sobre carreira                P2_p    
49              Parte2  Dados sobre carreira                P2_q    
50              Parte2  Dados sobre carreira                P2_r    
51              Parte2  Dados sobre carreira                P2_s    

                                              Questao pergunta  
16               Qual sua situação atual de trabalho?    P2_a   
17                                              Setor    P2_b   
18                             Numero de Funcionarios    P2_c   
19                                            Gestor?    P2_d   
20                                  Cargo como Gestor    P2_e   
21                                        Cargo Atual    P2_f   
22                                              Nivel    P2_g   
23                                     Faixa salarial    P2_h   
24  Quanto tempo de experiência na área de dados v...    P2_i   
25  Quanto tempo de experiência na área de TI/Enge...    P2_j   
26         Você está satisfeito na sua empresa atual?    P2_k   
27  Qual o principal motivo da sua insatisfação co...    P2_l   
28  Falta de oportunidade de crescimento no empreg...     P2_l  
29           Salário atual não corresponde ao mercado     P2_l  
30     Não tenho uma boa relação com meu líder/gestor     P2_l  
31  Gostaria de trabalhar em em outra área de atuação     P2_l  
32                Gostaria de receber mais benefícios     P2_l  
33             O clima de trabalho/ambiente não é bom     P2_l  
34           Falta de maturidade analítica na empresa     P2_l  
35  Você participou de entrevistas de emprego nos ...    P2_m   
36  Você pretende mudar de emprego nos próximos 6 ...    P2_n   
37  Quais os princi

In [56]:
descricao_filtrada = categoria[categoria['IdentificadorQuestao'].str.len() <= 5]
descricao_filtrada

DivisaoQuestionario         NomeCategoria IdentificadorQuestao  \
16              Parte2  Dados sobre carreira                P2_a    
17              Parte2  Dados sobre carreira                P2_b    
18              Parte2  Dados sobre carreira                P2_c    
19              Parte2  Dados sobre carreira                P2_d    
20              Parte2  Dados sobre carreira                P2_e    
21              Parte2  Dados sobre carreira                P2_f    
22              Parte2  Dados sobre carreira                P2_g    
23              Parte2  Dados sobre carreira                P2_h    
24              Parte2  Dados sobre carreira                P2_i    
25              Parte2  Dados sobre carreira                P2_j    
26              Parte2  Dados sobre carreira                P2_k    
27              Parte2  Dados sobre carreira                P2_l    
35              Parte2  Dados sobre carreira                P2_m    
36              Parte2  Dados sobre carreira                P2_n    
37              Parte2  Dados sobre carreira                P2_o    
48              Parte2  Dados sobre carreira                P2_p    
49              Parte2  Dados sobre carreira                P2_q    
50              Parte2  Dados sobre carreira                P2_r    
51              Parte2  Dados sobre carreira                P2_s    

                                              Questao pergunta  
16               Qual sua situação atual de trabalho?    P2_a   
17                                              Setor    P2_b   
18                             Numero de Funcionarios    P2_c   
19                                            Gestor?    P2_d   
20                                  Cargo como Gestor    P2_e   
21                                        Cargo Atual    P2_f   
22                                              Nivel    P2_g   
23                                     Faixa salarial    P2_h   
24  Quanto tempo de experiência na área de dados v...    P2_i   
25  Quanto tempo de experiência na área de TI/Enge...    P2_j   
26         Você está satisfeito na sua empresa atual?    P2_k   
27  Qual o principal motivo da sua insatisfação co...    P2_l   
35  Você participou de entrevistas de emprego nos ...    P2_m   
36  Você pretende mudar de emprego nos próximos 6 ...    P2_n   
37  Quais os principais critérios que você leva em...    P2_o   
48           Atualmente qual a sua forma de trabalho?    P2_p   
49          Qual a forma de trabalho ideal para você?    P2_q   
50  Caso sua empresa decida pelo modelo 100% prese...    P2_r   
51              Sua empresa passu por Layoff em 2022?    P2_s

In [57]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte3']
categoria

DivisaoQuestionario                            NomeCategoria  \
52              Parte3  Desafios dos gestores de times de dados   
53              Parte3  Desafios dos gestores de times de dados   
54              Parte3  Desafios dos gestores de times de dados   
55              Parte3  Desafios dos gestores de times de dados   
56              Parte3  Desafios dos gestores de times de dados   
57              Parte3  Desafios dos gestores de times de dados   
58              Parte3  Desafios dos gestores de times de dados   
59              Parte3  Desafios dos gestores de times de dados   
60              Parte3  Desafios dos gestores de times de dados   
61              Parte3  Desafios dos gestores de times de dados   
62              Parte3  Desafios dos gestores de times de dados   
63              Parte3  Desafios dos gestores de times de dados   
64              Parte3  Desafios dos gestores de times de dados   
65              Parte3  Desafios dos gestores de times de dados   
66              Parte3  Desafios dos gestores de times de dados   
67              Parte3  Desafios dos gestores de times de dados   
68              Parte3  Desafios dos gestores de times de dados   
69              Parte3  Desafios dos gestores de times de dados   
70              Parte3  Desafios dos gestores de times de dados   
71              Parte3  Desafios dos gestores de times de dados   
72              Parte3  Desafios dos gestores de times de dados   
73              Parte3  Desafios dos gestores de times de dados   
74              Parte3  Desafios dos gestores de times de dados   
75              Parte3  Desafios dos gestores de times de dados   
76              Parte3  Desafios dos gestores de times de dados   
77              Parte3  Desafios dos gestores de times de dados   
78              Parte3  Desafios dos gestores de times de dados   
79              Parte3  Desafios dos gestores de times de dados   
80              Parte3  Desafios dos gestores de times de dados   
81              Parte3  Desafios dos gestores de times de dados   
82              Parte3  Desafios dos gestores de times de dados   
83              Parte3  Desafios dos gestores de times de dados   
84              Parte3  Desafios dos gestores de times de dados   
85              Parte3  Desafios dos gestores de times de dados   
86              Parte3  Desafios dos gestores de times de dados   
87              Parte3  Desafios dos gestores de times de dados   
88              Parte3  Desafios dos gestores de times de dados   
89              Parte3  Desafios dos gestores de times de dados   

   IdentificadorQuestao                                            Questao  \
52                P3_a   Qual o número aproximado de pessoas que atuam ...   
53                P3_b   Quais desses papéis/cargos fazem parte do time...   
54              P3_b_1                                  Analytics Engineer   
55              P3_b_2                   Engenharia de Dados/Data Engineer   
56              P3_b_3                      Analista de Dados/Data Analyst   
57              P3_b_4                   Cientista de Dados/Data Scientist   
58              P3_b_5                          Database Administrator/DBA   
59              P3_b_6                Analista de Business Intelligence/BI   
60              P3_b_7                   Arquiteto de Dados/Data Architect   
61              P3_b_8                            Data Product Manager/DPM   
62              P3_b_9                                    Business Analyst   
63                P3_c   Quais dessas responsabilidades fazem parte da ...   
64              P3_c_1   Pensar na visão de longo prazo de dados da emp...   
65              P3_c_2   Organização de treinamentos e iniciativas com ...   
66              P3_c_3   Atração, seleção e contratação de talentos par...   
67              P3_c_4   Decisão sobre contratação de ferramentas e tec...   
68              P3_c_5   Sou gestor da equipe responsável pela 

In [58]:
descricao_filtrada = categoria[categoria['IdentificadorQuestao'].str.len() <= 5]
descricao_filtrada

DivisaoQuestionario                            NomeCategoria  \
52              Parte3  Desafios dos gestores de times de dados   
53              Parte3  Desafios dos gestores de times de dados   
63              Parte3  Desafios dos gestores de times de dados   
75              Parte3  Desafios dos gestores de times de dados   

   IdentificadorQuestao                                            Questao  \
52                P3_a   Qual o número aproximado de pessoas que atuam ...   
53                P3_b   Quais desses papéis/cargos fazem parte do time...   
63                P3_c   Quais dessas responsabilidades fazem parte da ...   
75                P3_d   Quais são os 3 maiores desafios que você tem c...   

   pergunta  
52    P3_a   
53    P3_b   
63    P3_c   
75    P3_d

In [59]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte4']
categoria

DivisaoQuestionario                   NomeCategoria IdentificadorQuestao  \
90               Parte4  Conhecimentos na área de dados                P4_a    
91               Parte4  Conhecimentos na área de dados              P4_a_1    
92               Parte4  Conhecimentos na área de dados                P4_b    
93               Parte4  Conhecimentos na área de dados              P4_b_1    
94               Parte4  Conhecimentos na área de dados              P4_b_2    
..                  ...                             ...                  ...   
185              Parte4  Conhecimentos na área de dados             P4_i_19    
186              Parte4  Conhecimentos na área de dados             P4_i_20    
187              Parte4  Conhecimentos na área de dados             P4_i_21    
188              Parte4  Conhecimentos na área de dados             P4_i_22    
189              Parte4  Conhecimentos na área de dados             P4_i_23    

                                               Questao pergunta  
90   Mesmo que esse não seja seu cargo formal, você...    P4_a   
91                                             Atuacao     P4_a  
92   Quais das fontes de dados listadas você já ana...    P4_b   
93      Dados relacionais (estruturados em bancos SQL)     P4_b  
94                   Dados armazenados em bancos NoSQL     P4_b  
..                                                 ...      ...  
185                                            Grafana     P4_i  
186                                     TIBCO Spotfire     P4_i  
187                                            Pentaho     P4_i  
188  Fazemos todas as análises utilizando apenas Ex...     P4_i  
189   Não utilizo nenhuma ferramenta de BI no trabalho     P4_i  

[100 rows x 5 columns]

In [60]:
descricao_filtrada = categoria[categoria['IdentificadorQuestao'].str.len() <= 5]
descricao_filtrada

DivisaoQuestionario                   NomeCategoria IdentificadorQuestao  \
90               Parte4  Conhecimentos na área de dados                P4_a    
92               Parte4  Conhecimentos na área de dados                P4_b    
101              Parte4  Conhecimentos na área de dados                P4_c    
110              Parte4  Conhecimentos na área de dados                P4_d    
125              Parte4  Conhecimentos na área de dados                P4_e    
126              Parte4  Conhecimentos na área de dados                P4_f    
127              Parte4  Conhecimentos na área de dados                P4_g    
161              Parte4  Conhecimentos na área de dados                P4_g    
162              Parte4  Conhecimentos na área de dados                P4_h    
166              Parte4  Conhecimentos na área de dados                P4_i    

                                               Questao pergunta  
90   Mesmo que esse não seja seu cargo formal, você...    P4_a   
92   Quais das fontes de dados listadas você já ana...    P4_b   
101  Entre as fontes de dados listadas, quais você ...    P4_c   
110  Quais das linguagens listadas abaixo você util...    P4_d   
125  Entre as linguagens listadas abaixo, qual é a ...    P4_e   
126  Entre as linguagens listadas abaixo, qual é a ...    P4_f   
127  Quais dos bancos de dados/fontes de dados list...    P4_g   
161  Quais das opções de Cloud listadas abaixo você...    P4_g   
162  Dentre as opções listadas, qual sua Cloud pref...    P4_h   
166                                  Microsoft PowerBI    P4_i

In [61]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte5']
categoria

DivisaoQuestionario               NomeCategoria IdentificadorQuestao  \
190              Parte5  Objetivos na área de dados                P5_a    
191              Parte5  Objetivos na área de dados                P5_b    
192              Parte5  Objetivos na área de dados                P5_c    
193              Parte5  Objetivos na área de dados                P5_d    

                                               Questao pergunta  
190                Qual seu objetivo na área de dados?    P5_a   
191              Qual oportunidade você está buscando?    P5_b   
192  Há quanto tempo você busca uma oportunidade na...    P5_c   
193  Como tem sido a busca por um emprego na área d...    P5_d

In [62]:
descricao_filtrada = categoria[categoria['IdentificadorQuestao'].str.len() <= 5]
descricao_filtrada

DivisaoQuestionario               NomeCategoria IdentificadorQuestao  \
190              Parte5  Objetivos na área de dados                P5_a    
191              Parte5  Objetivos na área de dados                P5_b    
192              Parte5  Objetivos na área de dados                P5_c    
193              Parte5  Objetivos na área de dados                P5_d    

                                               Questao pergunta  
190                Qual seu objetivo na área de dados?    P5_a   
191              Qual oportunidade você está buscando?    P5_b   
192  Há quanto tempo você busca uma oportunidade na...    P5_c   
193  Como tem sido a busca por um emprego na área d...    P5_d

In [63]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte6']
categoria

DivisaoQuestionario                            NomeCategoria  \
194              Parte6  Conhecimentos em Engenharia de Dados/DE   
195              Parte6  Conhecimentos em Engenharia de Dados/DE   
196              Parte6  Conhecimentos em Engenharia de Dados/DE   
197              Parte6  Conhecimentos em Engenharia de Dados/DE   
198              Parte6  Conhecimentos em Engenharia de Dados/DE   
199              Parte6  Conhecimentos em Engenharia de Dados/DE   
200              Parte6  Conhecimentos em Engenharia de Dados/DE   
201              Parte6  Conhecimentos em Engenharia de Dados/DE   
202              Parte6  Conhecimentos em Engenharia de Dados/DE   
203              Parte6  Conhecimentos em Engenharia de Dados/DE   
204              Parte6  Conhecimentos em Engenharia de Dados/DE   
205              Parte6  Conhecimentos em Engenharia de Dados/DE   
206              Parte6  Conhecimentos em Engenharia de Dados/DE   
207              Parte6  Conhecimentos em Engenharia de Dados/DE   
208              Parte6  Conhecimentos em Engenharia de Dados/DE   
209              Parte6  Conhecimentos em Engenharia de Dados/DE   
210              Parte6  Conhecimentos em Engenharia de Dados/DE   
211              Parte6  Conhecimentos em Engenharia de Dados/DE   
212              Parte6  Conhecimentos em Engenharia de Dados/DE   
213              Parte6  Conhecimentos em Engenharia de Dados/DE   
214              Parte6  Conhecimentos em Engenharia de Dados/DE   
215              Parte6  Conhecimentos em Engenharia de Dados/DE   
216              Parte6  Conhecimentos em Engenharia de Dados/DE   
217              Parte6  Conhecimentos em Engenharia de Dados/DE   
218              Parte6  Conhecimentos em Engenharia de Dados/DE   
219              Parte6  Conhecimentos em Engenharia de Dados/DE   
220              Parte6  Conhecimentos em Engenharia de Dados/DE   
221              Parte6  Conhecimentos em Engenharia de Dados/DE   
222              Parte6  Conhecimentos em Engenharia de Dados/DE   
223              Parte6  Conhecimentos em Engenharia de Dados/DE   
224              Parte6  Conhecimentos em Engenharia de Dados/DE   
225              Parte6  Conhecimentos em Engenharia de Dados/DE   
226              Parte6  Conhecimentos em Engenharia de Dados/DE   
227              Parte6  Conhecimentos em Engenharia de Dados/DE   
228              Parte6  Conhecimentos em Engenharia de Dados/DE   
229              Parte6  Conhecimentos em Engenharia de Dados/DE   
230              Parte6  Conhecimentos em Engenharia de Dados/DE   
231              Parte6  Conhecimentos em Engenharia de Dados/DE   
232              Parte6  Conhecimentos em Engenharia de Dados/DE   
233              Parte6  Conhecimentos em Engenharia de Dados/DE   
234              Parte6  Conhecimentos em Engenharia de Dados/DE   
235              Parte6  Conhecimentos em Engenharia de Dados/DE   
236              Parte6  Conhecimentos em Engenharia de Dados/DE   
237              Parte6  Conhecimentos em Engenharia de Dados/DE   
238              Parte6  Conhecimentos em Engenharia de Dados/DE   
239              Parte6  Conhecimentos em Engenharia de Dados/DE   
240              Parte6  Conhecimentos em Engenharia de Dados/DE   
241              Parte6  Conhecimentos em Engenharia de Dados/DE   
242              Parte6  Conhecimentos em Engenharia de Dados/DE   
243              Parte6  Conhecimentos em Engenharia de Dados/DE   
244              Parte6  Conhecimentos em Engenharia de Dados/DE   
245              Parte6  Conhecimentos em Engenharia de Dados/DE   
246              Parte6  Conhecimentos em Engenharia de Dados/DE   
247              Parte6  Conhecimentos em Engenharia de Dados/DE   
248              Parte6  Conhecimentos em Engenharia de Dados/DE   
249              Parte6  Conhecimentos em Engenharia de Dados/DE   
250              Parte6  Conhecimentos em Engenharia de Dados/DE   
251              Parte6  Conhecimentos em Engenharia de Dado

In [64]:
descricao_filtrada = categoria[categoria['IdentificadorQuestao'].str.len() <= 5]
descricao_filtrada

DivisaoQuestionario                            NomeCategoria  \
194              Parte6  Conhecimentos em Engenharia de Dados/DE   
204              Parte6  Conhecimentos em Engenharia de Dados/DE   
225              Parte6  Conhecimentos em Engenharia de Dados/DE   
226              Parte6  Conhecimentos em Engenharia de Dados/DE   
227              Parte6  Conhecimentos em Engenharia de Dados/DE   
228              Parte6  Conhecimentos em Engenharia de Dados/DE   
229              Parte6  Conhecimentos em Engenharia de Dados/DE   
243              Parte6  Conhecimentos em Engenharia de Dados/DE   

    IdentificadorQuestao                                            Questao  \
194                P6_a   Quais das opções abaixo fazem parte da sua rot...   
204                P6_b   Quais as ferramentas/tecnologias de ETL que vo...   
225                P6_c                Sua organização possui um Data Lake?   
226                P6_d   Qual tecnologia utilizada como plataforma do D...   
227                P6_e           Sua organização possui um Data Warehouse?   
228                P6_f   Qual tecnologia utilizada como plataforma do D...   
229                P6_g   Quais as ferramentas de gestão de Qualidade de...   
243                P6_h   Em qual das opções abaixo você gasta a maior p...   

    pergunta  
194    P6_a   
204    P6_b   
225    P6_c   
226    P6_d   
227    P6_e   
228    P6_f   
229    P6_g   
243    P6_h

## Problema na categoria 7 
253 	Parte7 	Conhecimentos em Análise de Dados/DA 	P7_1 	Quais das opções abaixo fazem parte da sua rot... 	P7_1

In [65]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte7']
categoria

DivisaoQuestionario                         NomeCategoria  \
253              Parte7  Conhecimentos em Análise de Dados/DA   
254              Parte7  Conhecimentos em Análise de Dados/DA   
255              Parte7  Conhecimentos em Análise de Dados/DA   
256              Parte7  Conhecimentos em Análise de Dados/DA   
257              Parte7  Conhecimentos em Análise de Dados/DA   
258              Parte7  Conhecimentos em Análise de Dados/DA   
259              Parte7  Conhecimentos em Análise de Dados/DA   
260              Parte7  Conhecimentos em Análise de Dados/DA   
261              Parte7  Conhecimentos em Análise de Dados/DA   
262              Parte7  Conhecimentos em Análise de Dados/DA   
263              Parte7  Conhecimentos em Análise de Dados/DA   
264              Parte7  Conhecimentos em Análise de Dados/DA   
265              Parte7  Conhecimentos em Análise de Dados/DA   
266              Parte7  Conhecimentos em Análise de Dados/DA   
267              Parte7  Conhecimentos em Análise de Dados/DA   
268              Parte7  Conhecimentos em Análise de Dados/DA   
269              Parte7  Conhecimentos em Análise de Dados/DA   
270              Parte7  Conhecimentos em Análise de Dados/DA   
271              Parte7  Conhecimentos em Análise de Dados/DA   
272              Parte7  Conhecimentos em Análise de Dados/DA   
273              Parte7  Conhecimentos em Análise de Dados/DA   
274              Parte7  Conhecimentos em Análise de Dados/DA   
275              Parte7  Conhecimentos em Análise de Dados/DA   
276              Parte7  Conhecimentos em Análise de Dados/DA   
277              Parte7  Conhecimentos em Análise de Dados/DA   
278              Parte7  Conhecimentos em Análise de Dados/DA   
279              Parte7  Conhecimentos em Análise de Dados/DA   
280              Parte7  Conhecimentos em Análise de Dados/DA   
281              Parte7  Conhecimentos em Análise de Dados/DA   
282              Parte7  Conhecimentos em Análise de Dados/DA   
283              Parte7  Conhecimentos em Análise de Dados/DA   
284              Parte7  Conhecimentos em Análise de Dados/DA   
285              Parte7  Conhecimentos em Análise de Dados/DA   
286              Parte7  Conhecimentos em Análise de Dados/DA   
287              Parte7  Conhecimentos em Análise de Dados/DA   
288              Parte7  Conhecimentos em Análise de Dados/DA   
289              Parte7  Conhecimentos em Análise de Dados/DA   
290              Parte7  Conhecimentos em Análise de Dados/DA   
291              Parte7  Conhecimentos em Análise de Dados/DA   
292              Parte7  Conhecimentos em Análise de Dados/DA   
293              Parte7  Conhecimentos em Análise de Dados/DA   
294              Parte7  Conhecimentos em Análise de Dados/DA   
295              Parte7  Conhecimentos em Análise de Dados/DA   
296              Parte7  Conhecimentos em Análise de Dados/DA   
297              Parte7  Conhecimentos em Análise de Dados/DA   
298              Parte7  Conhecimentos em Análise de Dados/DA   
299              Parte7  Conhecimentos em Análise de Dados/DA   
300              Parte7  Conhecimentos em Análise de Dados/DA   
301              Parte7  Conhecimentos em Análise de Dados/DA   
302              Parte7  Conhecimentos em Análise de Dados/DA   

    IdentificadorQuestao                                            Questao  \
253                P7_1   Quais das opções abaixo fazem parte da sua rot...   
254              P7_a_1   Processo e analiso dados utilizando linguagens...   
255              P7_a_2   Realizo construções de dashboards em ferrament...   
256              P7_a_3   Crio consultas através da linguagem SQL para e...   
257              P7_a_4                                         Utilizo API   
258              P7_a_5   Realizo experimentos e estudos utilizando meto...   
259              P7_a_6               Desenvolvo/cuido da manutenção de ETL   
260              P7_a_7   Atuo na modelagem dos dados, c

In [66]:
descricao_filtrada = categoria[categoria['IdentificadorQuestao'].str.len() <= 5]
descricao_filtrada

DivisaoQuestionario                         NomeCategoria  \
253              Parte7  Conhecimentos em Análise de Dados/DA   
264              Parte7  Conhecimentos em Análise de Dados/DA   
285              Parte7  Conhecimentos em Análise de Dados/DA   
292              Parte7  Conhecimentos em Análise de Dados/DA   

    IdentificadorQuestao                                            Questao  \
253                P7_1   Quais das opções abaixo fazem parte da sua rot...   
264                P7_b   Quais as ferramentas/tecnologias de ETL que vo...   
285                P7_c   Sua empresa utiliza alguma das ferramentas lis...   
292                P7_d   Em qual das opções abaixo você gasta a maior p...   

    pergunta  
253    P7_1   
264    P7_b   
285    P7_c   
292    P7_d

## Problema na categoria 8 
329 	Parte8 	Conhecimentos em Ciências de Dados/DS 	P8_3 	Quais dessas tecnologias fazem parte do seu di... 	P8_3

In [67]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte8']
categoria

DivisaoQuestionario                          NomeCategoria  \
303              Parte8  Conhecimentos em Ciências de Dados/DS   
304              Parte8  Conhecimentos em Ciências de Dados/DS   
305              Parte8  Conhecimentos em Ciências de Dados/DS   
306              Parte8  Conhecimentos em Ciências de Dados/DS   
307              Parte8  Conhecimentos em Ciências de Dados/DS   
308              Parte8  Conhecimentos em Ciências de Dados/DS   
309              Parte8  Conhecimentos em Ciências de Dados/DS   
310              Parte8  Conhecimentos em Ciências de Dados/DS   
311              Parte8  Conhecimentos em Ciências de Dados/DS   
312              Parte8  Conhecimentos em Ciências de Dados/DS   
313              Parte8  Conhecimentos em Ciências de Dados/DS   
314              Parte8  Conhecimentos em Ciências de Dados/DS   
315              Parte8  Conhecimentos em Ciências de Dados/DS   
316              Parte8  Conhecimentos em Ciências de Dados/DS   
317              Parte8  Conhecimentos em Ciências de Dados/DS   
318              Parte8  Conhecimentos em Ciências de Dados/DS   
319              Parte8  Conhecimentos em Ciências de Dados/DS   
320              Parte8  Conhecimentos em Ciências de Dados/DS   
321              Parte8  Conhecimentos em Ciências de Dados/DS   
322              Parte8  Conhecimentos em Ciências de Dados/DS   
323              Parte8  Conhecimentos em Ciências de Dados/DS   
324              Parte8  Conhecimentos em Ciências de Dados/DS   
325              Parte8  Conhecimentos em Ciências de Dados/DS   
326              Parte8  Conhecimentos em Ciências de Dados/DS   
327              Parte8  Conhecimentos em Ciências de Dados/DS   
328              Parte8  Conhecimentos em Ciências de Dados/DS   
329              Parte8  Conhecimentos em Ciências de Dados/DS   
330              Parte8  Conhecimentos em Ciências de Dados/DS   
331              Parte8  Conhecimentos em Ciências de Dados/DS   
332              Parte8  Conhecimentos em Ciências de Dados/DS   
333              Parte8  Conhecimentos em Ciências de Dados/DS   
334              Parte8  Conhecimentos em Ciências de Dados/DS   
335              Parte8  Conhecimentos em Ciências de Dados/DS   
336              Parte8  Conhecimentos em Ciências de Dados/DS   
337              Parte8  Conhecimentos em Ciências de Dados/DS   
338              Parte8  Conhecimentos em Ciências de Dados/DS   
339              Parte8  Conhecimentos em Ciências de Dados/DS   
340              Parte8  Conhecimentos em Ciências de Dados/DS   
341              Parte8  Conhecimentos em Ciências de Dados/DS   
342              Parte8  Conhecimentos em Ciências de Dados/DS   
343              Parte8  Conhecimentos em Ciências de Dados/DS   
344              Parte8  Conhecimentos em Ciências de Dados/DS   
345              Parte8  Conhecimentos em Ciências de Dados/DS   
346              Parte8  Conhecimentos em Ciências de Dados/DS   
347              Parte8  Conhecimentos em Ciências de Dados/DS   
348              Parte8  Conhecimentos em Ciências de Dados/DS   
349              Parte8  Conhecimentos em Ciências de Dados/DS   
350              Parte8  Conhecimentos em Ciências de Dados/DS   
351              Parte8  Conhecimentos em Ciências de Dados/DS   
352              Parte8  Conhecimentos em Ciências de Dados/DS   

    IdentificadorQuestao                                            Questao  \
303                P8_a   Quais das opções abaixo fazem parte da sua rot...   
304              P8_a_1   Estudos Ad-hoc com o objetivo de confirmar hip...   
305              P8_a_2   Sou responsável pela coleta e limpeza dos dado...   
306              P8_a_3   Sou responsável por entrar em contato com os t...   
307              P8_a_4   Desenvolvo modelos de Machine Learning com o o...   
308              P8_a_5   Sou responsável por colocar modelos em produçã...   
309              P8_a_6   Cuido da manutenção de modelos de Machine Lear...   
310  

In [68]:
descricao_filtrada = categoria[categoria['IdentificadorQuestao'].str.len() <= 5]
descricao_filtrada

DivisaoQuestionario                          NomeCategoria  \
303              Parte8  Conhecimentos em Ciências de Dados/DS   
315              Parte8  Conhecimentos em Ciências de Dados/DS   
329              Parte8  Conhecimentos em Ciências de Dados/DS   
341              Parte8  Conhecimentos em Ciências de Dados/DS   

    IdentificadorQuestao                                            Questao  \
303                P8_a   Quais das opções abaixo fazem parte da sua rot...   
315                P8_b   Quais as técnicas e métodos listados abaixo vo...   
329                P8_3   Quais dessas tecnologias fazem parte do seu di...   
341                P8_d   Em qual das opções abaixo você gasta a maior p...   

    pergunta  
303    P8_a   
315    P8_b   
329    P8_3   
341    P8_d

## tratando os problemas

In [71]:
descricao.loc[descricao['IdentificadorQuestao'] ==  'P7_1 ', 'IdentificadorQuestao'] = "P7_a "

In [73]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte7']
categoria

DivisaoQuestionario                         NomeCategoria  \
253              Parte7  Conhecimentos em Análise de Dados/DA   
254              Parte7  Conhecimentos em Análise de Dados/DA   
255              Parte7  Conhecimentos em Análise de Dados/DA   
256              Parte7  Conhecimentos em Análise de Dados/DA   
257              Parte7  Conhecimentos em Análise de Dados/DA   
258              Parte7  Conhecimentos em Análise de Dados/DA   
259              Parte7  Conhecimentos em Análise de Dados/DA   
260              Parte7  Conhecimentos em Análise de Dados/DA   
261              Parte7  Conhecimentos em Análise de Dados/DA   
262              Parte7  Conhecimentos em Análise de Dados/DA   
263              Parte7  Conhecimentos em Análise de Dados/DA   
264              Parte7  Conhecimentos em Análise de Dados/DA   
265              Parte7  Conhecimentos em Análise de Dados/DA   
266              Parte7  Conhecimentos em Análise de Dados/DA   
267              Parte7  Conhecimentos em Análise de Dados/DA   
268              Parte7  Conhecimentos em Análise de Dados/DA   
269              Parte7  Conhecimentos em Análise de Dados/DA   
270              Parte7  Conhecimentos em Análise de Dados/DA   
271              Parte7  Conhecimentos em Análise de Dados/DA   
272              Parte7  Conhecimentos em Análise de Dados/DA   
273              Parte7  Conhecimentos em Análise de Dados/DA   
274              Parte7  Conhecimentos em Análise de Dados/DA   
275              Parte7  Conhecimentos em Análise de Dados/DA   
276              Parte7  Conhecimentos em Análise de Dados/DA   
277              Parte7  Conhecimentos em Análise de Dados/DA   
278              Parte7  Conhecimentos em Análise de Dados/DA   
279              Parte7  Conhecimentos em Análise de Dados/DA   
280              Parte7  Conhecimentos em Análise de Dados/DA   
281              Parte7  Conhecimentos em Análise de Dados/DA   
282              Parte7  Conhecimentos em Análise de Dados/DA   
283              Parte7  Conhecimentos em Análise de Dados/DA   
284              Parte7  Conhecimentos em Análise de Dados/DA   
285              Parte7  Conhecimentos em Análise de Dados/DA   
286              Parte7  Conhecimentos em Análise de Dados/DA   
287              Parte7  Conhecimentos em Análise de Dados/DA   
288              Parte7  Conhecimentos em Análise de Dados/DA   
289              Parte7  Conhecimentos em Análise de Dados/DA   
290              Parte7  Conhecimentos em Análise de Dados/DA   
291              Parte7  Conhecimentos em Análise de Dados/DA   
292              Parte7  Conhecimentos em Análise de Dados/DA   
293              Parte7  Conhecimentos em Análise de Dados/DA   
294              Parte7  Conhecimentos em Análise de Dados/DA   
295              Parte7  Conhecimentos em Análise de Dados/DA   
296              Parte7  Conhecimentos em Análise de Dados/DA   
297              Parte7  Conhecimentos em Análise de Dados/DA   
298              Parte7  Conhecimentos em Análise de Dados/DA   
299              Parte7  Conhecimentos em Análise de Dados/DA   
300              Parte7  Conhecimentos em Análise de Dados/DA   
301              Parte7  Conhecimentos em Análise de Dados/DA   
302              Parte7  Conhecimentos em Análise de Dados/DA   

    IdentificadorQuestao                                            Questao  \
253                P7_a   Quais das opções abaixo fazem parte da sua rot...   
254              P7_a_1   Processo e analiso dados utilizando linguagens...   
255              P7_a_2   Realizo construções de dashboards em ferrament...   
256              P7_a_3   Crio consultas através da linguagem SQL para e...   
257              P7_a_4                                         Utilizo API   
258              P7_a_5   Realizo experimentos e estudos utilizando meto...   
259              P7_a_6               Desenvolvo/cuido da manutenção de ETL   
260              P7_a_7   Atuo na modelagem dos dados, c

In [74]:
descricao.loc[descricao['IdentificadorQuestao'] ==  'P8_3 ', 'IdentificadorQuestao'] = "P8_c "

In [75]:
categoria = descricao[descricao.DivisaoQuestionario=='Parte7']
categoria

DivisaoQuestionario                         NomeCategoria  \
253              Parte7  Conhecimentos em Análise de Dados/DA   
254              Parte7  Conhecimentos em Análise de Dados/DA   
255              Parte7  Conhecimentos em Análise de Dados/DA   
256              Parte7  Conhecimentos em Análise de Dados/DA   
257              Parte7  Conhecimentos em Análise de Dados/DA   
258              Parte7  Conhecimentos em Análise de Dados/DA   
259              Parte7  Conhecimentos em Análise de Dados/DA   
260              Parte7  Conhecimentos em Análise de Dados/DA   
261              Parte7  Conhecimentos em Análise de Dados/DA   
262              Parte7  Conhecimentos em Análise de Dados/DA   
263              Parte7  Conhecimentos em Análise de Dados/DA   
264              Parte7  Conhecimentos em Análise de Dados/DA   
265              Parte7  Conhecimentos em Análise de Dados/DA   
266              Parte7  Conhecimentos em Análise de Dados/DA   
267              Parte7  Conhecimentos em Análise de Dados/DA   
268              Parte7  Conhecimentos em Análise de Dados/DA   
269              Parte7  Conhecimentos em Análise de Dados/DA   
270              Parte7  Conhecimentos em Análise de Dados/DA   
271              Parte7  Conhecimentos em Análise de Dados/DA   
272              Parte7  Conhecimentos em Análise de Dados/DA   
273              Parte7  Conhecimentos em Análise de Dados/DA   
274              Parte7  Conhecimentos em Análise de Dados/DA   
275              Parte7  Conhecimentos em Análise de Dados/DA   
276              Parte7  Conhecimentos em Análise de Dados/DA   
277              Parte7  Conhecimentos em Análise de Dados/DA   
278              Parte7  Conhecimentos em Análise de Dados/DA   
279              Parte7  Conhecimentos em Análise de Dados/DA   
280              Parte7  Conhecimentos em Análise de Dados/DA   
281              Parte7  Conhecimentos em Análise de Dados/DA   
282              Parte7  Conhecimentos em Análise de Dados/DA   
283              Parte7  Conhecimentos em Análise de Dados/DA   
284              Parte7  Conhecimentos em Análise de Dados/DA   
285              Parte7  Conhecimentos em Análise de Dados/DA   
286              Parte7  Conhecimentos em Análise de Dados/DA   
287              Parte7  Conhecimentos em Análise de Dados/DA   
288              Parte7  Conhecimentos em Análise de Dados/DA   
289              Parte7  Conhecimentos em Análise de Dados/DA   
290              Parte7  Conhecimentos em Análise de Dados/DA   
291              Parte7  Conhecimentos em Análise de Dados/DA   
292              Parte7  Conhecimentos em Análise de Dados/DA   
293              Parte7  Conhecimentos em Análise de Dados/DA   
294              Parte7  Conhecimentos em Análise de Dados/DA   
295              Parte7  Conhecimentos em Análise de Dados/DA   
296              Parte7  Conhecimentos em Análise de Dados/DA   
297              Parte7  Conhecimentos em Análise de Dados/DA   
298              Parte7  Conhecimentos em Análise de Dados/DA   
299              Parte7  Conhecimentos em Análise de Dados/DA   
300              Parte7  Conhecimentos em Análise de Dados/DA   
301              Parte7  Conhecimentos em Análise de Dados/DA   
302              Parte7  Conhecimentos em Análise de Dados/DA   

    IdentificadorQuestao                                            Questao  \
253                P7_a   Quais das opções abaixo fazem parte da sua rot...   
254              P7_a_1   Processo e analiso dados utilizando linguagens...   
255              P7_a_2   Realizo construções de dashboards em ferrament...   
256              P7_a_3   Crio consultas através da linguagem SQL para e...   
257              P7_a_4                                         Utilizo API   
258              P7_a_5   Realizo experimentos e estudos utilizando meto...   
259              P7_a_6               Desenvolvo/cuido da manutenção de ETL   
260              P7_a_7   Atuo na modelagem dos dados, c

## Fazendo nova coluna

In [77]:
descricao_filtrada = descricao[descricao['IdentificadorQuestao'].str.len() <= 5]
descricao_filtrada

DivisaoQuestionario                          NomeCategoria  \
0                Parte0                          Identificador   
1                Parte1                     Dados demográficos   
3                Parte1                     Dados demográficos   
4                Parte1                     Dados demográficos   
5                Parte1                     Dados demográficos   
..                  ...                                    ...   
292              Parte7   Conhecimentos em Análise de Dados/DA   
303              Parte8  Conhecimentos em Ciências de Dados/DS   
315              Parte8  Conhecimentos em Ciências de Dados/DS   
329              Parte8  Conhecimentos em Ciências de Dados/DS   
341              Parte8  Conhecimentos em Ciências de Dados/DS   

    IdentificadorQuestao                                            Questao  \
0                     P0                                                 id   
1                  P1_a                                               Idade   
3                  P1_b                                              Genero   
4                  P1_c                                      Cor/raca/etnia   
5                  P1_d                                                 PCD   
..                   ...                                                ...   
292                P7_d   Em qual das opções abaixo você gasta a maior p...   
303                P8_a   Quais das opções abaixo fazem parte da sua rot...   
315                P8_b   Quais as técnicas e métodos listados abaixo vo...   
329                P8_c   Quais dessas tecnologias fazem parte do seu di...   
341                P8_d   Em qual das opções abaixo você gasta a maior p...   

    pergunta  
0         P0  
1      P1_a   
3      P1_b   
4      P1_c   
5      P1_d   
..       ...  
292    P7_d   
303    P8_a   
315    P8_b   
329    P8_3   
341    P8_d   

[66 rows x 5 columns]

In [87]:
dicionario = descricao_filtrada.set_index('pergunta')['Questao'].to_dict()

In [88]:
dicionario

{'P0': 'id',
 'P1_a ': 'Idade',
 'P1_b ': 'Genero',
 'P1_c ': 'Cor/raca/etnia',
 'P1_d ': 'PCD',
 'P1_e ': 'experiencia_profissional_prejudicada',
 'P1_f ': 'aspectos_prejudicados',
 'P1_g ': 'vive_no_brasil',
 'P1_i ': 'Estado onde mora',
 'P1_j ': 'Mudou de Estado?',
 'P1_k ': 'Regiao de origem',
 'P1_l ': 'Nivel de Ensino',
 'P1_m ': 'Área de Formação',
 'P2_a ': 'Qual sua situação atual de trabalho?',
 'P2_b ': 'Setor',
 'P2_c ': 'Numero de Funcionarios',
 'P2_d ': 'Gestor?',
 'P2_e ': 'Cargo como Gestor',
 'P2_f ': 'Cargo Atual',
 'P2_g ': 'Nivel',
 'P2_h ': 'Faixa salarial',
 'P2_i ': 'Quanto tempo de experiência na área de dados você tem?',
 'P2_j ': 'Quanto tempo de experiência na área de TI/Engenharia de Software você teve antes de começar a trabalhar na área de dados?',
 'P2_k ': 'Você está satisfeito na sua empresa atual?',
 'P2_l ': 'Qual o principal motivo da sua insatisfação com a empresa atual?',
 'P2_m ': 'Você participou de entrevistas de emprego nos últimos 6 meses?',